specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/preview/2024-03-01-preview/diagnostic.json

```diff
@@ -0,0 +1,695 @@
+        "id": {
+          "description": "Identifier for the type of diagnostic",
+          "type": "string"
+        },
+        "name": {
+          "description": "Friendly name for the type of diagnostic",
+          "type": "string"
+        },
+        "type": {
+          "description": "Type of the diagnostic",
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/DiagnosticProperties"
+        }
+      }
+    },
+    "DiagnosticProperties": {
+      "description": "Contains additional properties of a diagnostic",
+      "type": "object",
+      "properties": {
+        "supportedResourceTypes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Describes what are the supported resource types for a diagnostic."
+        }
+      }
+    },
+    "RunDiskInspectionInput": {
+      "type": "object",
+      "properties": {
+        "resourceId": {
+          "description": "Qualified ID of the resource.",
+          "type": "string"
+        },
+        "manifest": {
+          "description": "Name of manifest in order to trigger Disk Inspection.",
+          "type": "string"
+        },
+        "uploadSasUri": {
+          "description": "SAS uri to the blob where results will be uploaded.",
+          "type": "string",
+          "x-ms-secret": true
+        }
+      },
+      "required": [
+        "resourceId",
+        "manifest",
+        "uploadSasUri"
+      ],
+      "description": "Data used for requesting a Disk Inspection execution."
+    },
+    "ComputeDiagnosticsOperationResult": {
+      "type": "object",
+      "properties": {
+        "message": {
+          "type": "string",
+          "description": "The result of the disk inspection operation."
+        },
+        "responseFields": {
+          "type": "string",
+          "description": "The response fields of the disk inspection operation."
+        },
+        "resultStatus": {
+          "type": "string",
+          "enum": [
+            "Success",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "resultStatus",
+            "modelAsString": true
+          },
+          "description": "Result status of the async operation."
+        },
+        "errorDetail": {
+          "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorDetail",
+          "description": "The API error details."
+        },
+        "createdUTC": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The time when the disk inspection was completed."
+        }
+      },
+      "description": "Api output result when Compute Diagnostic operation is completed."      
+    },
+    "StorageConfigurationInput": {
+      "type": "object",
+      "properties": {
+        "storageAccountId": {
+          "description": "Qualified ID of the storage account.",
```

Comment:
>Qualified [](http://example.com/codeflow?start=26&length=9)

how is it qualified? Are we talking about something that is not a full Azure resource ID?

Reply:
Updated description with example

Reply:
Oh, we could call it 'the resource ID' or 'Azure resource ID' then